monitoring/src/config.ts

```diff
@@ -106,19 +85,73 @@ const availableEnvConfig = [
 	ConfigAusCode,
 ];
 
-export const envConfig: Config = (() => {
-	const jurisdiction = decideJurisdiction(envJurisdiction, envAwsRegion);
-	const foundConfig = availableEnvConfig.find(
-		(value) =>
-			value.stage == envStage.toLowerCase() &&
-			value.jurisdiction == jurisdiction,
-	);
-
-	if (foundConfig === undefined) {
-		const j = envJurisdiction ?? 'missing';
-		const r = envAwsRegion ?? 'missing';
-		throw `No config found for (env)stage: ${envStage}, (env)jurisdiction: ${j}, (env)aws-region: ${r}`;
+export class ConfigWrapper {
+	private _jurisdiction: JurisdictionOpt;
+	private _stage: string;
+	private _awsRegion: AwsRegionOpt;
+	private _config: Config | undefined;
+
+	get stage(): string {
+		return this._stage;
+	}
+
+	get jurisdiction(): JurisdictionOpt {
+		return this._jurisdiction;
+	}
+
+	get awsRegion(): AwsRegionOpt {
+		return this._awsRegion;
 	}
 
-	return foundConfig;
-})();
+	get config(): Config | undefined {
+		return this._config;
+	}
+
+	set config(value: Config | undefined) {
+		this._config = value;
+	}
+
+	constructor(
+		_envAwsRegion: AwsRegionOpt = envAwsRegion,
+		_envStage: string = envStage,
+		_envJurisdiction: JurisdictionOpt = envJurisdiction,
+	) {
+		this._jurisdiction = _envJurisdiction;
+		this._awsRegion = _envAwsRegion;
+		this._stage = _envStage;
+	}
+
+	async run(): Promise<void> {
+		if (this._config) {
+			await this._config.checkFunction(this._config);
+		}
+	}
+
+	public generateConfig(): void {
+		// If no jurisdiction assign using aws region (Scheduled)
+		if (!this._jurisdiction && this._awsRegion) {
+			this._jurisdiction = ConfigHelper.getJurisdiction(this._awsRegion);
+			log_info(`Generating config for scheduled trigger`);
+		}
+
+		// If no aws Region assign using jurisdiction (Adhoc)
```

Comment:
This makes me think we should probably be offering the region choice up front in the UI _not_ jurisdiction. Then the later issue of assigning region from jurisdiction goes away.